azext_iot/tests/dps/device_registration/test_iot_device_registration_individual_int.py

```diff
@@ -4,521 +4,465 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-import json
+from azext_iot.common.embedded_cli import EmbeddedCLI
 from azext_iot.common.shared import EntityStatusType, AttestationType
 from azext_iot.tests.dps import (
     DATAPLANE_AUTH_TYPES,
-    IoTDPSLiveScenarioTest
+    clean_dps_dataplane
 )
-from azext_iot.tests.dps.device_registration import compare_registrations
-from azext_iot.tests.helpers import CERT_ENDING, KEY_ENDING
-
-
-class TestDPSDeviceRegistrationsIndividual(IoTDPSLiveScenarioTest):
-    def __init__(self, test_case):
-        super(TestDPSDeviceRegistrationsIndividual, self).__init__(test_case, cert_only=False)
-        self.id_scope = self.get_dps_id_scope()
-
-    def test_dps_device_registration_symmetrickey_lifecycle(self):
-        attestation_type = AttestationType.symmetricKey.value
-        hub_host_name = f"{self.entity_hub_name}.azure-devices.net"
-        for auth_phase in DATAPLANE_AUTH_TYPES:
-            enrollment_id = self.generate_enrollment_names()[0]
-            device_id = self.generate_device_names()[0]
-
-            # Enrollment needs to be created
-            self.cmd(
-                self.set_cmd_auth_type(
-                    "iot device registration create --dps-name {} -g {} --registration-id {}".format(
-                        self.entity_dps_name, self.entity_rg, enrollment_id
-                    ),
-                    auth_type=auth_phase
-                ),
-                expect_failure=True
+from azext_iot.tests.dps.device_registration import compare_registrations, check_hub_device
+from azext_iot.tests.helpers import CERT_ENDING, KEY_ENDING, create_test_cert, set_cmd_auth_type
+from azext_iot.tests.generators import generate_names
+
+cli = EmbeddedCLI()
+
+
+def test_dps_device_registration_symmetrickey_lifecycle(provisioned_iot_dps_module):
```

Comment:
yes? not really sure what you are asking for here.

Every test is a function so that we do not need to rely on attributes and methods from base classes. Also conftest does not play the best with classes (I am sure there is way to force it but this is easier).

Reply:
"Also conftest does not play the best with classes" I got the same experience as well, just asked due to curiosity. Thanks!